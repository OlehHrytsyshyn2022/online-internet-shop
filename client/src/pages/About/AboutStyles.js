import { makeStyles } from "@mui/styles";

export const useStyles = makeStyles(() => ({
  pageContainer: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    maxWidth: 1200,
  },

  article1: {
    display: "inline-block",
    paddingTop: 50,
    fontSize: 16,
    fontWeight: 400,
    lineHeight: "27px",
    paddingBottom: 50,
    maxWidth: 670,
    textAlign: "left",
  },
  article2: {
    display: "inline-block",
    fontSize: 16,
    fontWeight: 400,
    lineHeight: "27px",
    paddingBottom: 50,
    maxWidth: 670,
    textAlign: "left",
  },
}));
