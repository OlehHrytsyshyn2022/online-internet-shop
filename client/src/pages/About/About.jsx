import React from "react";
import { Typography, Container, IconButton, Divider } from "@mui/material";
import { Facebook, Twitter, Instagram } from "@mui/icons-material";
import { useStyles } from "./AboutStyles";

function About() {
  const classes = useStyles();

  const shareUrl = "https://shoppe.com"; // треба замінити на URL свого веб-сайту

  return (
    <Container maxWidth="lg" className={classes.pageContainer}>
      {/* Перше фото */}
      <Typography variant="h4" sx={{ marginTop: "50px" }}>
        About us
      </Typography>
      <Typography variant="h6" sx={{ marginBottom: "25px" }}>
        Fast Fashion, And Faster Fashion
      </Typography>
      <img
        src="https://res.cloudinary.com/ddh4awlkr/image/upload/v1693001757/online-internet-shop/about/Img_01_yzmci9.jpg"
        alt="Перше фото"
        style={{ width: "100%", borderRadius: "8px", objectFit: "cover" }}
      />
      <Typography variant="body1" className={classes.article1}>
        To create a unique image, you use all means to create a unique look, so
        sometimes it is simply necessary to buy jewelry. It is gold and silver
        jewelry that helps you show your individuality, impeccable taste, boost
        your self-confidence, and reveal your femininity.
        <br />
        <br />
        Jewelry made of silver and gold goes well with almost any color and skin
        tone. They can be bought and worn by people of any age and gender. Even
        large, massive pieces made of these noble metals have a graceful,
        elegant look. To create a unique image, all means are used, so sometimes
        it is simply necessary to buy jewelry. It is gold and silver jewelry
        that helps to show your individuality, impeccable taste, boosts
        self-confidence, and reveals femininity.
        <br />
        <br />
        Jewelry perfectly emphasizes your personality. They can be bought and
        worn by people of any age and gender. Even large, massive pieces made of
        this noble metal have a graceful, elegant look. Our online store closely
        follows fashion trends in the world. Even large, massive pieces of
        jewelry have a graceful, elegant look.
      </Typography>

      {/* Друге фото  */}

      <img
        src="https://res.cloudinary.com/ddh4awlkr/image/upload/v1693001756/online-internet-shop/about/Img_02_e8mw2w.jpg"
        alt="Друге фото"
        style={{
          display: "block",
          margin: "0 auto",
          width: "58%",
          minWidth: "288px",
          minHeight: "149px",
          borderRadius: "8px",
          objectFit: "cover",
        }}
      />

      {/* Текст після другого фото */}

      <Typography variant="body1" className={classes.article2}>
        <Typography
          component="span"
          sx={{
            display: "block",
            paddingTop: "50px",
            paddingBottom: "20px",
            fontSize: "26px",
            fontWeight: 400,
          }}
        >
          TOP TRENDS
        </Typography>
        Individual approach to each client We offer only proven and high-quality
        products. Interesting and stylish models for you. We carefully examine
        each piece of jewelry. Only then do we put it up for sale.
        <br />
        <br />
        Each item has a sample from the State Assay Office of Ukraine and
        international quality certificates. Convenient, fast order form.
        Individual approach to each client. Our website has a lot of
        fashionable, stylish and inexpensive jewelry that you won't be able to
        find and buy in city stores. You will be unique and inimitable. Our
        online jewelry store offers to buy only high-quality, interesting models
        of products.
      </Typography>

      {/* Кнопка для Фейсбуку */}

      <Container
        sx={{
          display: "flex",
          justifyContent: "center",
          padding: "0",
          marginBottom: "100px",
        }}
      >
        <Typography sx={{ marginTop: "10px", fontSize: "16px" }}>
          Share
        </Typography>
        <Divider
          flexItem
          sx={{ flexGrow: 1, marginLeft: "8px", maxWidth: "500px" }}
        />
        <IconButton
          aria-label="Поділитися на Фейсбук"
          component="a"
          href={`https://www.facebook.com/sharer/sharer.php?u=${shareUrl}`}
          target="_blank"
        >
          <Facebook />
        </IconButton>

        {/* Кнопка для Твіттера */}

        <IconButton
          aria-label="Поділитися на Твіттері"
          component="a"
          href={`https://twitter.com/intent/tweet?url=${shareUrl}`}
          target="_blank"
        >
          <Twitter />
        </IconButton>

        {/* Кнопка для Інстаграму */}
        <IconButton
          aria-label="Поділитися на Інстаграмі"
          component="a"
          href={`https://www.instagram.com/${shareUrl}`}
          target="_blank"
        >
          <Instagram />
        </IconButton>
      </Container>
    </Container>
  );
}
export default About;
