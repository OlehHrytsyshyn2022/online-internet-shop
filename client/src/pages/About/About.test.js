import React from "react";
import { render } from "@testing-library/react";
import About from "./About";

describe("About", () => {
  test("renders About page", () => {
    const { getByText } = render(<About />);
    const AboutText = getByText("About us");
    expect(AboutText).toBeInTheDocument();
  });

  test("Snapshot Test About", () => {
    const { container } = render(<About />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
