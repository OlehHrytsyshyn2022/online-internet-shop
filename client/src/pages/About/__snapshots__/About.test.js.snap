// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Snapshot Test About 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg makeStyles-pageContainer-4 css-1oqqzyl-MuiContainer-root"
>
  <h4
    class="MuiTypography-root MuiTypography-h4 css-16z2ulp-MuiTypography-root"
  >
    About us
  </h4>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-1wl10xm-MuiTypography-root"
  >
    Fast Fashion, And Faster Fashion
  </h6>
  <img
    alt="Перше фото"
    src="https://res.cloudinary.com/ddh4awlkr/image/upload/v1693001757/online-internet-shop/about/Img_01_yzmci9.jpg"
    style="width: 100%; border-radius: 8px; object-fit: cover;"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 makeStyles-article1-5 css-ahj2mt-MuiTypography-root"
  >
    To create a unique image, you use all means to create a unique look, so sometimes it is simply necessary to buy jewelry. It is gold and silver jewelry that helps you show your individuality, impeccable taste, boost your self-confidence, and reveal your femininity.
    <br />
    <br />
    Jewelry made of silver and gold goes well with almost any color and skin tone. They can be bought and worn by people of any age and gender. Even large, massive pieces made of these noble metals have a graceful, elegant look. To create a unique image, all means are used, so sometimes it is simply necessary to buy jewelry. It is gold and silver jewelry that helps to show your individuality, impeccable taste, boosts self-confidence, and reveals femininity.
    <br />
    <br />
    Jewelry perfectly emphasizes your personality. They can be bought and worn by people of any age and gender. Even large, massive pieces made of this noble metal have a graceful, elegant look. Our online store closely follows fashion trends in the world. Even large, massive pieces of jewelry have a graceful, elegant look.
  </p>
  <img
    alt="Друге фото"
    src="https://res.cloudinary.com/ddh4awlkr/image/upload/v1693001756/online-internet-shop/about/Img_02_e8mw2w.jpg"
    style="display: block; margin: 0px auto; width: 58%; min-width: 288px; min-height: 149px; border-radius: 8px; object-fit: cover;"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 makeStyles-article2-6 css-ahj2mt-MuiTypography-root"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 css-gjod31-MuiTypography-root"
    >
      TOP TRENDS
    </span>
    Individual approach to each client We offer only proven and high-quality products. Interesting and stylish models for you. We carefully examine each piece of jewelry. Only then do we put it up for sale.
    <br />
    <br />
    Each item has a sample from the State Assay Office of Ukraine and international quality certificates. Convenient, fast order form. Individual approach to each client. Our website has a lot of fashionable, stylish and inexpensive jewelry that you won't be able to find and buy in city stores. You will be unique and inimitable. Our online jewelry store offers to buy only high-quality, interesting models of products.
  </p>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-85nmtf-MuiContainer-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-5uatv5-MuiTypography-root"
    >
      Share
    </p>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-1663a1l-MuiDivider-root"
    />
    <a
      aria-label="Поділитися на Фейсбук"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      href="https://www.facebook.com/sharer/sharer.php?u=https://shoppe.com"
      tabindex="0"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FacebookIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      aria-label="Поділитися на Твіттері"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      href="https://twitter.com/intent/tweet?url=https://shoppe.com"
      tabindex="0"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="TwitterIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      aria-label="Поділитися на Інстаграмі"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      href="https://www.instagram.com/https://shoppe.com"
      tabindex="0"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="InstagramIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
